### PR TITLE
Task 23: FE: Aside filter component. Implement categories filter

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/components/Layout/Categories/Categories.tsx [hunk 1]
```diff
@@ -1,28 +1,76 @@
-import { useEffect } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import { AppDispatch } from "../../../redux/app/store";
 import { useDispatch } from "react-redux";
 import { useAppSelector } from "../../../main/hooks";
 import { getAllCategories } from "../../../redux/features/categories/categoriesSlice";
```
Comment (VYaniv) [inline src/components/Layout/Categories/Categories.tsx:5]: Don`t you have aliases?

File: src/components/Layout/Categories/Categories.tsx [hunk 2]
```diff
@@ -1,28 +1,76 @@
...
+import { Popover } from "../../Popover/Popover";
 
 export const Categories = () => {
   const dispatch: AppDispatch = useDispatch();
   const { categories } = useAppSelector((state) => state.categories);
 
+  const [activeCategoryId, setActiveCategoryId] = useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null,
+  );
```
Comment (VYaniv) [inline src/components/Layout/Categories/Categories.tsx:15]: Unneeded empty line. Direct import please. Keep consistency.

File: src/components/Layout/Categories/Categories.tsx [hunk 3]
```diff
@@ -1,28 +1,76 @@
...
+            className="flex gap-1 items-center cursor-pointer"
+            key={category._id}
+          >
+            <div className="text-black font-medium">{category.name}</div>
+            <img
+              src="/vector.svg"
+              alt="Vector icon bottom"
+              className="w-2.5 h-2.5"
+            />
+          </button>
```
Comment (VYaniv) [inline src/components/Layout/Categories/Categories.tsx:54]: It even can be a separate component

**Review by VYaniv — APPROVED** [review]

File: src/components/Layout/Categories/Categories.tsx (hunk 2 above)
Comment (di-kinosian) [reply src/components/Layout/Categories/Categories.tsx:15]: done

**Pull request merged**